src/config.c

```diff
@@ -3238,6 +3317,7 @@ standardConfig static_configs[] = {
     createSpecialConfig("bind", NULL, MODIFIABLE_CONFIG | MULTI_ARG_CONFIG, setConfigBindOption, getConfigBindOption, rewriteConfigBindOption, applyBind),
     createSpecialConfig("replicaof", "slaveof", IMMUTABLE_CONFIG | MULTI_ARG_CONFIG, setConfigReplicaOfOption, getConfigReplicaOfOption, rewriteConfigReplicaOfOption, NULL),
     createSpecialConfig("latency-tracking-info-percentiles", NULL, MODIFIABLE_CONFIG | MULTI_ARG_CONFIG, setConfigLatencyTrackingInfoPercentilesOutputOption, getConfigLatencyTrackingInfoPercentilesOutputOption, rewriteConfigLatencyTrackingInfoPercentilesOutputOption, NULL),
+    createSpecialConfig("trusted-addresses", NULL, MODIFIABLE_CONFIG | MULTI_ARG_CONFIG, setConfigTrustedAddresses, getConfigTrustedAddresses, rewriteConfigTrustedAddresses, NULL),
```

Comment:
Ok, I really don't like special configs. I think there is a much better way to implement the exact same thing.

```
void updateTrustedAddresses(char *err) {
    convertTrustedAddressesIntoIPs(server.trusted_addresses_string);
}

    createStringConfig("trusted-addresses", NULL, MODIFIABLE_CONFIG, EMPTY_STRING_IS_NULL, server.trusted_addresses_string, NULL, NULL, updateTrustedAddresses),
```

Then you have a single piece of logic that converts the raw string version into the underlying data structure we use at runtime. We never need to deal with the logic to convert the underlying representation back to a string and don't have to implement custom rewrite logic.

Reply:
When you check the definition of createStringConfig, you will find the functions:  stringConfigInit, stringConfigSet, stringConfigGet, stringConfigRewrite only deal with one string, they can not handle the string array,  this is why I choose the createSpecialConfig, it allows us to customize how to deal with config get/set/rewrite operations.